### PR TITLE
SM and SMZ3 option_definitions deprecation fix

### PR DESCRIPTION
**Pull request by lordlou**
## What is this fixing or adding?
Replaced deprecated use of option_definitions for SM and SMZ3 by options_dataclass

## How was this tested?
Generated a multiworld with both game that succeeded without having warning message about deprecated use of option_definitions.

### CODE REVIEW

**Comment by Exempt-Medic** [comment]
You'll also want to change `get_filler_item_name` in sm's `__init__.py`

File: worlds/sm/variaRandomizer/utils/utils.py [hunk 1]
```diff
@@ -358,35 +358,35 @@ def convertParam(randoParams, param, inverse=False):
...
-        args.superFun.append("Combat")
-    if world.fun_movement[player].value:
+        args.areaLayoutBase = not options.area_layout.value
+    args.escapeRando = options.escape_rando.value
+    args.noRemoveEscapeEnemies = not options.remove_escape_enemies.value
+    args.doorsColorsRando = options.doors_colors_rando.value
+    args.allowGreyDoors = options.allow_grey_doors.value
+    args.bosses = options.boss_randomization.value
+    if options.fun_combat.value:
+        args.superFun.append()
```
Comment (Exempt-Medic) [inline worlds/sm/variaRandomizer/utils/utils.py:386]: I think this was removed accidentally
```suggestion
        args.superFun.append("Combat")
```

**Review by Exempt-Medic — APPROVED** [review]
Changes LGTM. Generations (Ten sets of 100 fully random yamls) before and after were identical. No warning about deprecated option getters, and searches for others came up empty

**Review by qwint — APPROVED** [review]
format of the new options API looks good, did some searching and did not find any other uses of the old metadata.option_name[player] format
generated with default options and skip_output and confirmed there was no sm/smz3 warnings about Options API
did not test option combinations or fully patching and playing the relevant games

**Comment by Exempt-Medic** [comment]
This has conflicts btw @lordlou

File: worlds/smz3/Options.py [hunk 2]
```diff
@@ -126,20 +127,19 @@ class EnergyBeep(DefaultOnToggle):
...
+    goal: Goal
+    key_shuffle: KeyShuffle
+    open_tower: OpenTower
+    ganon_vulnerable: GanonVulnerable
+    open_tourian: OpenTourian
+    spin_jumps_animation: SpinJumpsAnimation
+    heart_beep_speed: HeartBeepSpeed
+    heart_color: HeartColor
+    quick_swap: QuickSwap
+    energy_beep: EnergyBeep
```
Comment (Exempt-Medic) [inline worlds/smz3/Options.py:145]: EoF newline
```suggestion
    energy_beep: EnergyBeep

```

File: worlds/smz3/__init__.py [hunk 3]
```diff
@@ -244,7 +246,7 @@ def set_rules(self):
             set_rule(entrance, lambda state, region=region: region.CanEnter(state.smz3state[self.player]))
             for loc in region.Locations:
                 l = self.locations[loc.Name]
-                if self.multiworld.accessibility[self.player] != 'full':
+                if self.multiworld.accessibility[self.player].value != Accessibility.option_full:
```
Comment (Exempt-Medic) [inline worlds/smz3/__init__.py:249]: Should be this:
```suggestion
                if self.options.accessibility.value != ItemsAccessibility.option_full:
```

**Review by Exempt-Medic — APPROVED** [review]
Reaffirming approval after the recent changes

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**